src/core/Object3D.js

```diff
@@ -567,165 +567,59 @@ THREE.Object3D.prototype = {
-
-		};
-
-		//
-
-		var parseObject = function ( object ) {
-
-			var data = {};
-
-			data.uuid = object.uuid;
-			data.type = object.type;
-
-			if ( object.name !== '' ) data.name = object.name;
-			if ( JSON.stringify( object.userData ) !== '{}' ) data.userData = object.userData;
-			if ( object.visible !== true ) data.visible = object.visible;
-
-			if ( object instanceof THREE.PerspectiveCamera ) {
-
-				data.fov = object.fov;
-				data.aspect = object.aspect;
-				data.near = object.near;
-				data.far = object.far;
-
-			} else if ( object instanceof THREE.OrthographicCamera ) {
-
-				data.left = object.left;
-				data.right = object.right;
-				data.top = object.top;
-				data.bottom = object.bottom;
-				data.near = object.near;
-				data.far = object.far;
-
-			} else if ( object instanceof THREE.AmbientLight ) {
-
-				data.color = object.color.getHex();
-
-			} else if ( object instanceof THREE.DirectionalLight ) {
-
-				data.color = object.color.getHex();
-				data.intensity = object.intensity;
-
-			} else if ( object instanceof THREE.PointLight ) {
-
-				data.color = object.color.getHex();
-				data.intensity = object.intensity;
-				data.distance = object.distance;
-				data.decay = object.decay;
-
-			} else if ( object instanceof THREE.SpotLight ) {
-
-				data.color = object.color.getHex();
-				data.intensity = object.intensity;
-				data.distance = object.distance;
-				data.angle = object.angle;
-				data.exponent = object.exponent;
-				data.decay = object.decay;
-
-			} else if ( object instanceof THREE.HemisphereLight ) {
-
-				data.color = object.color.getHex();
-				data.groundColor = object.groundColor.getHex();
-
-			} else if ( object instanceof THREE.Mesh || object instanceof THREE.Line || object instanceof THREE.PointCloud ) {
-
-				data.geometry = parseGeometry( object.geometry );
-				data.material = parseMaterial( object.material );
-
-				if ( object instanceof THREE.Line ) data.mode = object.mode;
-
-			} else if ( object instanceof THREE.Sprite ) {
-
-				data.material = parseMaterial( object.material );
-
+				generator: 'Object3D.toJSON'
 			}
 
-			data.matrix = object.matrix.toArray();
+	  }
 
-			if ( object.children.length > 0 ) {
+	  // standard Object3D serialization
 
-				data.children = [];
+	  data.type = this.type;
+	  data.uuid = this.uuid;
+	  if ( this.name !== '' ) data.name = this.name;
+	  if ( JSON.stringify( this.userData ) !== '{}' ) data.userData = this.userData;
+	  if ( this.visible !== true ) data.visible = this.visible;
 
-				for ( var i = 0; i < object.children.length; i ++ ) {
+	  data.matrix = this.matrix.toArray();
 
-					data.children.push( parseObject( object.children[ i ] ) );
+	  if ( this.children.length > 0 ) {
 
-				}
+	    data.children = [];
 
-			}
+	    for ( var i = 0; i < this.children.length; i ++ ) {
 
-			return data;
+	      data.children.push( this.children[ i ].toJSON( meta ) );
 
-		}
+	    }
 
-		output.object = parseObject( this );
```

Comment:
The new way outputs the object at the top level instead of at the property object.

The easiest way to solve this is by creating a new object as meta.object and then assign data as that object.